### PR TITLE
Add bubble sort algorithm

### PR DESCRIPTION
**Pull request by walmer26**
I Have added a tester module to the project for better test (skipping doctest).
Bubble sort has passed the test. Please do final review.

### CODE REVIEW

File: sorts/bubble_sort.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+def bubble_sort(unorder_list: list) -> list:
+    """
+    Bubble sort, sometimes referred to as sinking sort, is a simple sorting algorithm that repeatedly steps through the list, compares adjacent elements and swaps them if they are in the wrong order. The pass through the list is repeated until the list is sorted. The algorithm, which is a comparison sort, is named for the way smaller or larger elements "bubble" to the top of the list.
+    """
+    count = len(unorder_list) # sets the lenght of the array as a count so next iteractions is sorter as it is reduced by one each cycle.
+    while count >= 2: # initiates outter iteration based on array lenght 'count'.
+        for i in range(count): # inner iteration goes through each array set by outter loop.
+            temp_i = None # a temporary variable to hold an item of the pair is a swap is required.
+            if i < count-1 and unorder_list[i] > unorder_list[i+1]: # check if a swap is requited and execute the swap.
```
Comment (rayan147) [inline sorts/bubble_sort.py:9]: if  the array is mostly sorted  or sorted , an you avoid unnecessary  iterations.

**Review by rayan147 — CHANGES_REQUESTED** [review]
Added the review comments below . Also, Can you please rebase base your commit to just one commit  `chore: add bubble sort ` and change the branch's name to issue/2  . I was not specific  with the branch's name

File: sorts/bubble_sort.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+                temp_i = unorder_list[i+1]
+                unorder_list[i+1] = unorder_list[i]
+                unorder_list[i] = temp_i
+        count -= 1 # reduces the lenght of the array when the inner loop ends to start a next cucle with a sorter array.
+    return unorder_list
```
[the remark this answers is not in the document]
Comment (walmer26) [reply sorts/bubble_sort.py]: done

File: tester.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    def test_bubble_sort(self):
+        """testing bubble sort"""
+        self.assertEqual(
+            self.bsfn.bubble_sort(self.test_list),
+            self.bubble_sort_result
+        )
+
+if __name__ == "__main__":
+    unittest.main()
```
[the remark this answers is not in the document]
Comment (walmer26) [reply tester.py:19]: Thanks

File: sorts/bubble_sort.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    Bubble sort, sometimes referred to as sinking sort, is a simple sorting algorithm that repeatedly steps through the list, compares adjacent elements and swaps them if they are in the wrong order. The pass through the list is repeated until the list is sorted. The algorithm, which is a comparison sort, is named for the way smaller or larger elements "bubble" to the top of the list.
+    """
+    count = len(unorder_list) # sets the lenght of the array as a count so next iteractions is sorter as it is reduced by one each cycle.
+    while count >= 2: # initiates outter iteration based on array lenght 'count'.
+        for i in range(count): # inner iteration goes through each array set by outter loop.
+            temp_i = None # a temporary variable to hold an item of the pair is a swap is required.
+            if i < count-1 and unorder_list[i] > unorder_list[i+1]: # check if a swap is requited and execute the swap.
+                temp_i = unorder_list[i+1]
+                unorder_list[i+1] = unorder_list[i]
+                unorder_list[i] = temp_i
```
[the remark this answers is not in the document]
Comment (walmer26) [reply sorts/bubble_sort.py]: As swapped in one line, no function required, please review new bubble sort code.

File: sorts/bubble_sort.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+def bubble_sort(unorder_list: list) -> list:
```
[the remark this answers is not in the document]
Comment (walmer26) [reply sorts/bubble_sort.py:1]: Thanks

File: sorts/bubble_sort.py (hunk 1 above)
Comment (walmer26) [reply sorts/bubble_sort.py]: Please review new code additions. unnecessary iterations were eliminated.

**Comment by rayan147** [comment]
Perfect

**Pull request merged**